allennlp/common/testing/model_test_case.py

```diff
@@ -215,6 +215,9 @@ def ensure_batch_predictions_are_consistent(self):
                     # Loss is particularly unstable; we'll just be satisfied if everything else is
                     # close.
                     continue
+                if 'batch_weight' == key:
```

Comment:
doing yoda-style comparisons doesn't feel idiomatic in python (and what they protect against is an error in python already)

```
In [1]: x = 10

In [2]: if x = 1: print("oops")
  File "<ipython-input-2-380718274afc>", line 1
    if x = 1: print("oops")
         ^
SyntaxError: invalid syntax
```

Reply:
also, again I don't like the "magical" behavior here. is there a solution where `ensure_batch_predictions_are_consistent` has an optional arg for keys to ignore, and then the relevant test can just specify that?